recipe/activate.sh

```diff
@@ -1,10 +1,26 @@
 #!/bin/bash
-if [ "$(uname)" == "Darwin" ]
-then
-    # for Mac OSX
-    export KERAS_BACKEND=tensorflow
-elif [ "$(uname)" == "Linux" ]
-then
-    # for Linux
-    export KERAS_BACKEND=theano
+
+# get the location of the keras.json file
+KERAS_JSON="$HOME/.keras/keras.json"
+
+if [ -f $KERAS_JSON ]; then # if the file exists
+
+  BACKEND_KEY="backend" # value we are interested in finding
+  KEY_VALUE_REGEX="\"($BACKEND_KEY)\": \"([^\"]*)\"" # regex
+  JSON_TEXT=$(cat $KERAS_JSON) # data in keras.json file
+
+  if [[ $JSON_TEXT =~ $KEY_VALUE_REGEX ]]; then # check for a match
+    BACKEND_VALUE="${BASH_REMATCH[2]}" # get the user defined backend value
+    export KERAS_BACKEND=$BACKEND_VALUE # set the environmental variable
+  fi
```

Comment:
I think this was the curcial piece that I (both of us?) were missing...

> You can also define the environment variable `KERAS_BACKEND` and this will override what is defined in your config file

IOW even if you had the configuration file set correctly it would be overwritten by the environment variable set by the activation script. One could the environment variable each time after activation, but that is definitely obnoxious and puts into question the value of the activation script.

Thus I can definitely see this becomes a huge pain in terms of typical usage and with any proposed fix that still keeps the activation script in play.

Before I would say I was maybe a -0.5 with the activation script. In light of this new revelation, I'm becoming a -1 with the activation script.

Thoughts @croth1?